tests/test_contact_specific_functions.py

```diff
@@ -352,69 +263,51 @@ def test_calculate_contact_forces_self_rod():
 
     "Testing function with analytically verified values"
 
-    tol = 1e-5
-
-    "initializing rod parameters"
-    rod_position_collection = np.array(
-        [[1, 2, 3, 4, 5], [6, 7, 8, 9, 10], [11, 12, 13, 14, 15]]
-    )
-    rod_radius_collection = np.array(
-        [5.12952263, 0.19493561, 0.72904683, 0.21676892, 0.18705351]
-    )
-    rod_length_collection = np.array(
-        [0.89050271, 0.7076813, 0.21078658, 0.95372826, 0.86037329]
+    rod = MockRod()
+    """Increasing rod elements and length to establish self contact in rod;
+    elements are placed such that they are penetrating the next to next element by 1.0 unit in the x-direction."""
+    rod.n_elems = 4
+    rod.position_collection = np.array([[1, 3, 5, 7], [0, 0, 0, 0], [0, 0, 0, 0]])
+    rod.radius_collection = np.array([1, 1, 1, 1])
+    rod.length_collection = np.array([3.0, 3.0, 3.0, 3.0])
```

Comment:
Make sure that each element length is equal to the distances between the two adjacent nodes, looks like the spacing between the nodes is 2 above but the length are 3 here